File: part-02/src/main/java/eu/luminis/workshop/smallsteps/logic/domainmodel/valueobjects/LegoStoreId.java
package eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects;

import java.util.UUID;

public class LegoStoreId {
    public static final String PREFIX = "stor_";
    private final UUID storeId;

    public LegoStoreId() {
        this(UUID.randomUUID());
    }

    public LegoStoreId(String storeId) {
        this(UUID.fromString(storeId.replaceFirst("^" + PREFIX, "")));
    }

    public LegoStoreId(UUID storeId) {
        this.storeId = storeId;
    }

    @Override
    public String toString() {
        return PREFIX + this.storeId;
    }
}

File: part-02/src/main/kotlin/eu/luminis/workshop/smallsteps/logic/domainModel/valueObjects/LegoStoreId.kt
package eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects

import java.util.*

data class LegoStoreId(val id: UUID = UUID.randomUUID()) {
    constructor(storeId: String) : this(UUID.fromString(storeId.removePrefix(PREFIX)))

    override fun toString() = "$PREFIX$id"

    companion object {
        const val PREFIX = "stor_"
    }
}